halo2_proofs/src/poly.rs

```diff
@@ -209,6 +209,82 @@ impl<F: Field> Polynomial<F, LagrangeCoeff> {
             _marker: PhantomData,
         }
     }
+
+    /// Gets the specified chunk of the rotated version of this polynomial.
+    ///
+    /// Equivalent to:
+    /// ```ignore
+    /// self.rotate(rotation)
+    ///     .chunks(chunk_size)
+    ///     .nth(chunk_index)
+    ///     .unwrap()
+    ///     .to_vec()
+    /// ```
+    pub(crate) fn get_chunk_of_rotated(
+        &self,
+        rotation: Rotation,
+        chunk_size: usize,
+        chunk_index: usize,
+    ) -> Vec<F> {
```

Comment:
Could we avoid copying for the case where the chunk doesn't span a rotation boundary, by using `Cow` or something? (waves hands)

Reply:
We'd need to alter the recursive AST evaluation to operate over `Cow` as well, and within it we basically always will be allocating. This would only remove allocations at the leaf layers, and then only for chunks that don't overlap the wrapping boundary (which to be fair is most of them). But that still might be noticeable; we could try that in a subsequent PR.

Reply:
I gave this a go locally, and saw no effect on the Orchard circuit. I'll still open it as a separate PR after this one for others to test, in case it is still worthwhile.